src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts

```diff
@@ -0,0 +1,1066 @@
+					"outcome": { "classification": "SystemMetaData", "purpose": "FeatureInsight" }
+				}
+			*/
+			this.telemetryService.publicLog(strings.installedEvent, {
+				from: telemetryFrom,
+				extensionId: extensionSuggestion.id,
+				outcome: isPromiseCanceledError(err) ? 'canceled' : 'error',
+			});
+			this.notificationService.error(err);
+		});
+	}
+
+	private updateInstalledExtensions(container: HTMLElement, installedExtensions: Promise<IExtensionStatus[]>) {
+		installedExtensions.then(extensions => {
+			const elements = container.querySelectorAll('.installExtension, .enabledExtension');
+			for (let i = 0; i < elements.length; i++) {
+				elements[i].classList.remove('installed');
+			}
+			extensions.filter(ext => ext.globallyEnabled)
+				.map(ext => ext.identifier.id)
+				.forEach(id => {
+					const install = container.querySelectorAll(`.installExtension[data-extension="${id}"]`);
+					for (let i = 0; i < install.length; i++) {
+						install[i].classList.add('installed');
+					}
+					const enabled = container.querySelectorAll(`.enabledExtension[data-extension="${id}"]`);
+					for (let i = 0; i < enabled.length; i++) {
+						enabled[i].classList.add('installed');
+					}
+				});
+		}).then(undefined, onUnexpectedError);
+	}
+}
+
+export class WelcomeInputFactory implements IEditorInputFactory {
+
+	static readonly ID = welcomeInputTypeId;
+
+	public canSerialize(editorInput: EditorInput): boolean {
+		return true;
+	}
+
+	public serialize(editorInput: EditorInput): string {
+		return '{}';
+	}
+
+	public deserialize(instantiationService: IInstantiationService, serializedEditorInput: string): WalkThroughInput {
+		return instantiationService.createInstance(WelcomePage)
+			.editorInput;
+	}
+}
+
+// theming
+
+export const buttonBackground = registerColor('welcomePage.buttonBackground', { dark: null, light: null, hc: null }, localize('welcomePage.buttonBackground', 'Background color for the buttons on the Welcome page.'));
+export const buttonHoverBackground = registerColor('welcomePage.buttonHoverBackground', { dark: null, light: null, hc: null }, localize('welcomePage.buttonHoverBackground', 'Hover background color for the buttons on the Welcome page.'));
+export const welcomePageBackground = registerColor('welcomePage.background', { light: null, dark: null, hc: null }, localize('welcomePage.background', 'Background color for the Welcome page.'));
+
+registerThemingParticipant((theme, collector) => {
+
+	const backgroundColor = theme.getColor(welcomePageBackground);
+	if (backgroundColor) {
+		collector.addRule(`.monaco-workbench .part.editor > .content .welcomePageContainer { background-color: ${backgroundColor}; }`);
+	}
+	const tileColor = theme.getColor(tileBackground);
+	if (tileColor) {
+		collector.addRule(`.monaco-workbench .part.editor > .content .welcomePageContainer .tile:not(.extension):not(.extension_pack) { background-color: ${tileColor};  }`);
```

Comment:
Out of curiosity, any reason these aren't in a CSS file?

Reply:
This was at Anthony's request, that all colors should be in the registry for theming purposes.